diff --git a/xiaohongshu/login.go b/xiaohongshu/login.go
index 67e002d..4569f9e 100644
--- a/xiaohongshu/login.go
+++ b/xiaohongshu/login.go
@@ -23 +23,2 @@ func (a *LoginAction) CheckLoginStatus(ctx context.Context) (bool, error) {
-	pp.MustNavigate("https://www.xiaohongshu.com/explore").MustWaitLoad()
+	pp.MustNavigate("https://www.xiaohongshu.com/explore")
+	pp.MustWait(`() => document.querySelector(".main-container .user .link-wrapper .channel") !== null || document.querySelector(".login-container .qrcode-img") !== null`)
@@ -25 +25,0 @@ func (a *LoginAction) CheckLoginStatus(ctx context.Context) (bool, error) {
-	time.Sleep(1 * time.Second)
@@ -94 +94,2 @@ func (a *LoginAction) FetchQrcodeImage(ctx context.Context) (string, bool, error
-	pp.MustNavigate("https://www.xiaohongshu.com/explore").MustWaitLoad()
+	pp.MustNavigate("https://www.xiaohongshu.com/explore")
+	pp.MustWait(`() => document.querySelector(".main-container .user .link-wrapper .channel") !== null || document.querySelector(".login-container .qrcode-img") !== null`)
@@ -96 +96,0 @@ func (a *LoginAction) FetchQrcodeImage(ctx context.Context) (string, bool, error
-	time.Sleep(2 * time.Second)
